contributors/design-proposals/node/cri-windows.md

```diff
@@ -0,0 +1,65 @@
+# CRI: Windows Container Configuration
+
+**Authors**: Jiangtian Li (@JiangtianLi), Pengfei Ni (@feiskyer)
+
+**Status**: Proposed
+
+## Background
+Container Runtime Interface (CRI) defines [APIs and configuration types](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto) for kubelet to integrate various container runtimes. The container configuration is platform dependent. For example, on Linux platform, cpu quota and cpu period represent CPU resource allocation to tasks in a cgroup and cgroup by [Linux kernel CFS scheduler](https://www.kernel.org/doc/Documentation/scheduler/sched-design-CFS.txt). While on Windows platform, cpu shares represents [CPU rate control information](https://msdn.microsoft.com/en-us/library/windows/desktop/hh448384(v=vs.85).aspx) for a job object by Windows kernel scheduler. Open Container Initiative (OCI) Runtime Specification defines [platform specific configuration](https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration), including Linux, Windows, and Solaris. Currently CRI only suppports Linux container configuration.
+
+## Umbrella Issue
+[#56734](https://github.com/kubernetes/kubernetes/issues/56734)
+
+## Motivation
+The goal is to fill the gap of platform support in CRI, specifically for Windows platform. For example, currrently in dockershim Windows containers are scheduled using the default resource constraints and does not respect the resource requests and limits specified in POD. With this proposal, Windows containers will be able to leverage POD spec and CRI to allocate compute resource and respect restriction.
```

Comment:
It'd be nice to know besides resources, what other parts of the current API is lacking to support Windows containers (e.g., security context?).

Reply:
@yujuhong According to https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L418-L434, I think there are a few other properties such as CredentialSpec or HyperV that could be useful and already implemented in Windows. How does LinuxContainerSecurityContext map to https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L132-L164? I saw some similarity such as Seccomp. Selinux and Apparmor is in Process struct though (https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L30-L57)

@taylorb-microsoft Please advise what property besides Resources is needed to support Windows containers now (https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L418-L434).

Reply:
@yujuhong @JiangtianLi I have a draft on this that we're discussing in SIG-Windows. @feiskyer  also has a draft that we're merging together for the SIG. Next meeting is 9 Jan. After that we'll send it up to SIG-Node

Reply:
@PatrickLang  feel free to email me the draft and i can take a look ahead of our jan9 meeting

Reply:
> Next meeting is 9 Jan. After that we'll send it up to SIG-Node

Thanks for the update! I think we can hold on to this PR and wait for the more complete proposal to be presented in both SIGs.